pkg/server/workerconn.go

```diff
@@ -107,7 +107,10 @@ func (wc *WorkerConnection) Handle() {
 			return
 		}
 
-		command, message := proto.ChompCommand(data)
+		// Copy our data out
+		msg := make([]byte, n)
+		copy(msg, buffer[:n])
+		command, message := proto.ChompCommand(msg)
```

Comment:
I like the data length model you added in fossil and would love that here as well eventually.

Reply:
yeah, maybe i'll work on porting that over :)